docs/rfcs/rfc-034-generic-operators.md

```diff
@@ -0,0 +1,332 @@
+
+    """
+    Return Posts where all of the related Users match this filter
+    """
+    likes_ALL: UserWhere
+
+    """
+    Return Posts where none of the related Users match this filter
+    """
+    likes_NONE: UserWhere
+
+    """
+    Return Posts where one of the related Users match this filter
+    """
+    likes_SINGLE: UserWhere
+
+    """
+    Return Posts where some of the related Users match this filter
+    """
+    likes_SOME: UserWhere
+    likesConnection: PostLikesConnectionWhere @deprecated(reason: "Use `likesConnection_SOME` instead.")
+    likesConnection_NOT: PostLikesConnectionWhere @deprecated(reason: "Use `likesConnection_NONE` instead.")
+    likesConnection_ALL: PostLikesConnectionWhere
+    likesConnection_NONE: PostLikesConnectionWhere
+    likesConnection_SINGLE: PostLikesConnectionWhere
+    likesConnection_SOME: PostLikesConnectionWhere
+}
+```
+
+As it's visible from the schema generated, the current approach is to add operators in the schema in the form of:
+
+`{fieldName}_{operatorName}`: `{operatorType}`
+
+to the types needed.
+
+This mechanism is adopted widely, for instance, consider the aggregation filters:
+
+```graphql
+input PostLikesNodeAggregationWhereInput {
+    AND: [PostLikesNodeAggregationWhereInput!]
+    OR: [PostLikesNodeAggregationWhereInput!]
+    NOT: PostLikesNodeAggregationWhereInput
+    name_EQUAL: String
+    name_AVERAGE_EQUAL: Float
+    name_LONGEST_EQUAL: Int
+    name_SHORTEST_EQUAL: Int
+    name_GT: Int
+    name_AVERAGE_GT: Float
+    name_LONGEST_GT: Int
+    name_SHORTEST_GT: Int
+    name_GTE: Int
+    name_AVERAGE_GTE: Float
+    name_LONGEST_GTE: Int
+    name_SHORTEST_GTE: Int
+    name_LT: Int
+    name_AVERAGE_LT: Float
+    name_LONGEST_LT: Int
+    name_SHORTEST_LT: Int
+    name_LTE: Int
+    name_AVERAGE_LTE: Float
+    name_LONGEST_LTE: Int
+    name_SHORTEST_LTE: Int
+    someFloat_EQUAL: Float
+    someFloat_AVERAGE_EQUAL: Float
+    someFloat_MIN_EQUAL: Float
+    someFloat_MAX_EQUAL: Float
+    someFloat_SUM_EQUAL: Float
+    someFloat_GT: Float
+    someFloat_AVERAGE_GT: Float
+    someFloat_MIN_GT: Float
+    someFloat_MAX_GT: Float
+    someFloat_SUM_GT: Float
+    someFloat_GTE: Float
+    someFloat_AVERAGE_GTE: Float
+    someFloat_MIN_GTE: Float
+    someFloat_MAX_GTE: Float
+    someFloat_SUM_GTE: Float
+    someFloat_LT: Float
+    someFloat_AVERAGE_LT: Float
+    someFloat_MIN_LT: Float
+    someFloat_MAX_LT: Float
+    someFloat_SUM_LT: Float
+    someFloat_LTE: Float
+    someFloat_AVERAGE_LTE: Float
+    someFloat_MIN_LTE: Float
+    someFloat_MAX_LTE: Float
+    someFloat_SUM_LTE: Float
+}
+```
+
+And then these can be used as follow:
+
+**Posts where content contains Matrix**
+
+```graphql
+query Posts {
+    posts(where: { content_CONTAINS: "The Matrix" }) {
+        content
+    }
+}
+```
+
+**Posts with more than 10 likes**
+
+```graphql
+query Posts {
+    posts(where: { likesAggregate: { count_GT: 10 } }) {
+        content
+    }
+}
+```
+
```

Comment:
Is it worth putting the problem the current type defs that you are trying to solve?

I assume it's to reduce the size of our type defs & to have inputs that are more similar to other tools/APIs so it is easier for new users to get to grips with the library but I think it'd be good to have it spelled out

Reply:
I think it's more the improve the readability and to be fair is because is nicer, but from a pragmatic point of view, unfortunately until GraphQL does not support Generics, it will result in a larger number of types!
I still believe that consistency and readability should have priority on this!